app/courses/courses.component.ts

```diff
@@ -10,12 +11,17 @@ import {FilterPipe} from '../filter.pipe';
 })
 
 export class CoursesComponent {
+    private alive: boolean = true;
     courses: Course[];
 
     constructor(private courseService: CourseService, private filter: FilterPipe) {}
 
     getCourses(): void {
-        this.courses = this.courseService.getList();
+        this.courseService.getList()
+            .takeWhile(() => this.alive)
```

Comment:
it's better to explicitly subscribe and unsubscribe than to perform such checks. although such check would work, it's a little bit cryptive to use such form